backend/src/main/java/com/redhat/cloud/notifications/oapi/OApiFilter.java

```diff
@@ -0,0 +1,203 @@
+package com.redhat.cloud.notifications.oapi;
+
+import com.redhat.cloud.notifications.Constants;
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.mutiny.core.Vertx;
+import io.vertx.mutiny.ext.web.client.WebClient;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.WebApplicationException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class OApiFilter {
+
+    public static final String NOTIFICATIONS = "notifications";
+    public static final String INTEGRATIONS = "integrations";
+    public static final String PRIVATE = "private";
+    public static final String INTERNAL = "internal";
+
+    static List<String> whats = List.of(INTEGRATIONS, NOTIFICATIONS, PRIVATE, INTERNAL);
+
+    @Inject
+    Vertx vertx;
+
+    @ConfigProperty(name = "quarkus.http.port", defaultValue = "8085")
+    Integer port;
+
+    private WebClient client;
+
+    @PostConstruct
+    void initialize() {
+        this.client = WebClient.create(vertx,
+                new WebClientOptions().setDefaultHost("localhost").setDefaultPort(port));
+    }
+
+    public Uni<String> serveOpenApi(String what) {
```

Comment:
same here :)

Reply:
Same as above, will update to a better name